### PR TITLE
add username to admin search

### PR DESCRIPTION
**Pull request by carl-codeorg**
Adds username as a searchable field on the admin search page.

Screenshot 2024-03-27 at 9 15 30 AM

## Links

[JIRA](https://codedotorg.atlassian.net/browse/P20-725)

## Testing story

Tested locally

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/admin_search_controller.rb [hunk 1]
```diff
@@ -15,6 +15,9 @@ def find_students
     users = User.with_deleted
 
     # If requested, filter...
+    if params[:usernameFilter].present?
+      users = users.where("username LIKE ?", "%#{params[:usernameFilter]}%")
```
Comment (artem-vavilov) [inline dashboard/app/controllers/admin_search_controller.rb]: Please use `sanitize_sql_like` to filter potential SQL injections in the param:
```suggestion
      users = users.where("username LIKE ?", "%#{sanitize_sql_like(params[:usernameFilter])}%")
```

Or, I would recommend using ActiveRecord Arels:
```suggestion
      users = users.where(User.arel_table[:username].matches("%#{params[:usernameFilter]}%"))
```

Comment (artem-vavilov) [reply dashboard/app/controllers/admin_search_controller.rb]: Note: Leading wildcards (e.g., '%TERM') prevent the use of the field index for this query.

**Review by mgc1194 — APPROVED** [review]

**Review by mgc1194 — COMMENTED** [review]
Looks good. Could you update the [tests](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/admin_search_controller_test.rb) too?

**Review by artem-vavilov — APPROVED** [review]
Nice work! 🙌🏻

**Pull request merged**